core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -175,6 +206,16 @@ public String concreteSingleIndex(ClusterState state, IndicesRequest request) {
     }
 
     /**
+     * @return whether the specified alias or index exists. If the alias or index contains datemath then that is resolved too.
+     */
+    public boolean hasIndexOrAliasAndResolveDateMath(String aliasOrIndex, ClusterState state) {
```

Comment:
naming is hard.... :)  I find hasIndexOrAlias enough, we can then document that it will also resolve date math expressions?